client/jetpack-connect/authorize.js

```diff
@@ -793,8 +793,15 @@ export class JetpackAuthorize extends Component {
 			return translate( 'Return to your site' );
 		}
 
-		if ( this.isWooPasswordlessJPC() ) {
-			return translate( 'Connect your account' );
+		if ( config.isEnabled( 'woocommerce/core-profiler-passwordless-auth' ) ) {
```

Comment:
Small suggestion to avoid the eslint disable 
```suggestion
if (this.isWooPasswordlessJPC()) {
    if (config.isEnabled('woocommerce/core-profiler-passwordless-auth')) {
        return translate('Connect to WordPress.com');
    } else {
        return translate('Connect your account');
    }
}
```